subprojects/wrapper-shared/src/main/java/org/gradle/util/internal/ZipSlip.java

```diff
@@ -47,6 +47,6 @@ public static boolean isUnsafeZipEntryName(String name) {
     }
 
     private static boolean isWindows() {
-        return System.getProperty("os.name").contains("windows");
+        return System.getProperty("os.name").toLowerCase().contains("windows");
```

Comment:
I think that change is good. I am still wondering if instead of just rejecting a path with a forbidden character, we could also calculate the canonicalized path in that case in check if it is in the target directory.